graph.go

```diff
@@ -149,6 +149,29 @@ func (g *Graph) Node(id string) Node {
 	return n
 }
 
+// DeleteNode deletes a node and all the edges associated to the node
+func (g *Graph) DeleteNode(id string) error {
+	if _, ok := g.findNode(id); ok {
+		// Remove Node
+		delete(g.nodes, id)
+		// Remove all the edges from the Node
+		delete(g.edgesFrom, id)
+		fmt.Printf("%+v", g.edgesFrom)
+		// Remove all the edges to the Node
+		for parent, edgeList := range g.edgesFrom {
+			for i, edge := range edgeList {
+				if edge.to.id == id {
```

Comment:
looks like this can be done more efficiently; break the loop after deletion

Reply:
I'm not sure about this one, because two nodes A & B can have several edges going from A to B (happens in my project). The complexity is bad (the average is the number of edges...)  but I don't think there is another way to do so, and removing a node should be a one-time operation, not something to be optimized. But that's your lib, don't hesitate to tell me what you think about this :)

Reply:
you are probably right.  and again, the whole lib is not build for performance.